server/src/main/java/org/apache/druid/curator/discovery/CuratorDruidLeaderSelector.java

```diff
@@ -105,29 +105,30 @@ public void isLeader()
             }
             catch (Exception ex) {
               log.makeAlert(ex, "listener becomeLeader() failed. Unable to become leader").emit();
-              stopAndCreateNewLeaderLatch();
-              startLeaderLatch();
+              notLeader();
             }
           }
 
           @Override
           public void notLeader()
           {
-            try {
-              if (!leader) {
-                log.warn("I'm being asked to stop being leader. But I am not the leader. Ignored event.");
-                return;
-              }
+            if (!leader) {
+              log.warn("I'm being asked to stop being leader. But I am not the leader. Ignored event.");
+              return;
+            }
+
+            // Stop the current latch and create a new one
+            leader = false;
+            closeLeaderLatchQuietly();
 
-              leader = false;
-              // give others a chance to become leader.
-              stopAndCreateNewLeaderLatch();
+            try {
               listener.stopBeingLeader();
-              startLeaderLatch();
             }
             catch (Exception ex) {
               log.makeAlert(ex, "listener.stopBeingLeader() failed. Unable to stopBeingLeader").emit();
```

Comment:
When this happens, the OL is left in an inconsistent, partially-leading state and nothing will again ever fix it. It seems dangerous to leave the OL in this state. Perhaps we should `System.exit(1)` in this case so we can start back up and try to get back into a normal state. I notice the `K8sDruidLeaderSelector` already does this.

Reply:
Done.